### PR TITLE
Feature/design and limits

### PR DESCRIPTION
**Pull request by hihilla**

### CODE REVIEW

File: Pods/EPContactsPicker.swift [hunk 1]
```diff
@@ -15,13 +15,19 @@ public protocol EPPickerDelegate: class {
     func epContactPicker(_: EPContactsPicker, didCancel error: NSError)
     func epContactPicker(_: EPContactsPicker, didSelectContact contact: EPContact)
 	func epContactPicker(_: EPContactsPicker, didSelectMultipleContacts contacts: [EPContact])
+  
+  func enableSendButton(enabled: Bool, selectedContacts: [EPContact])
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: indentation

File: Pods/EPContactsPicker.swift [hunk 2]
```diff
@@ -15,13 +15,19 @@ public protocol EPPickerDelegate: class {
...
+  func presentAlert()
 }
 
 public extension EPPickerDelegate {
 	func epContactPicker(_: EPContactsPicker, didContactFetchFailed error: NSError) { }
 	func epContactPicker(_: EPContactsPicker, didCancel error: NSError) { }
 	func epContactPicker(_: EPContactsPicker, didSelectContact contact: EPContact) { }
 	func epContactPicker(_: EPContactsPicker, didSelectMultipleContacts contacts: [EPContact]) { }
+  
+  func enableSendButton(enabled: Bool, selectedContacts: [EPContact]) { }
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: indentation

File: Pods/EPContactsPicker.swift [hunk 3]
```diff
@@ -153,19 +178,7 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
...
-                let okAction = UIAlertAction(title: "Ok", style: UIAlertActionStyle.default, handler: {  action in
-                    completion([], error)
-                    self.dismiss(animated: true, completion: {
-                        self.contactDelegate?.epContactPicker(self, didContactFetchFailed: error)
-                    })
-                })
-                alert.addAction(okAction)
-                self.present(alert, animated: true, completion: nil)
-            
+                self.contactDelegate?.presentAlert()
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: I think the name of this method is too generic, maybe presentContactEditScreen?

File: Pods/EPContactsPicker.swift [hunk 4]
```diff
@@ -233,6 +254,75 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
...
+    let heightConstraint = NSLayoutConstraint(item: textField, attribute: .height, relatedBy: .equal,
+                                              toItem: nil, attribute: .notAnAttribute, multiplier: 1, constant: 25)
+    textField.addConstraint(heightConstraint)
+  }
+  
+  fileprivate func addContactAction(_ alertController: UIAlertController) {
+    let nameField = (alertController.textFields![0] as UITextField).text ?? ""
+    let phoneField = (alertController.textFields![1] as UITextField).text ?? ""
+    
+    if nameField != "" && phoneField != "" {
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: I would: 
(1) remove " ?? "" " from the test fields extraction 
(2) use guard to see if they're nil

File: Pods/EPContactsPicker.swift [hunk 5]
```diff
@@ -233,6 +254,75 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
...
+      // create a new contact
+      let contact = CNMutableContact()
+      contact.givenName = nameField
+      let phoneNumber = CNLabeledValue(label: CNLabelPhoneNumberiPhone,
+                                       value: CNPhoneNumber(stringValue: phoneField))
+      contact.phoneNumbers.append(phoneNumber)
+      
+      let key = self.sortedContactKeys.first
+      self.orderedContacts[key!]?.insert(contact, at: 1)
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: Use guard

File: Pods/EPContactsPicker.swift [hunk 6]
```diff
@@ -233,6 +254,75 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
...
+    } else {
+      let errorAlert = UIAlertController(title: "Error", message: "Please input name AND phone number", preferredStyle: .alert)
+      errorAlert.addAction(UIAlertAction(title: "OK", style: .cancel, handler: { alert -> Void in
+        self.present(alertController, animated: true, completion: nil)
+      }))
+      self.present(errorAlert, animated: true, completion: nil)
+    }
+  }
+  
+  func newContact() {
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: presenetNewContactScreen?

File: Pods/EPContactsPicker.swift [hunk 7]
```diff
@@ -270,16 +360,26 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
 		
         if multiSelectEnabled  && selectedContacts.contains(where: { $0.contactId == contact.contactId }) {
             cell.accessoryType = UITableViewCellAccessoryType.checkmark
+          cell.tintColor = EPGlobalConstants.Colors.nxYellow
         }
 		
         cell.updateContactsinUI(contact, indexPath: indexPath, subtitleType: subtitleCellValue)
+      if indexPath.row == 0 && indexPath.section == 0 {
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: indentation

File: Pods/EPContactsPicker.swift [hunk 8]
```diff
@@ -270,16 +360,26 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
 		
         if multiSelectEnabled  && selectedContacts.contains(where: { $0.contactId == contact.contactId }) {
             cell.accessoryType = UITableViewCellAccessoryType.checkmark
+          cell.tintColor = EPGlobalConstants.Colors.nxYellow
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: indentation

File: Pods/EPContactsPicker.swift [hunk 9]
```diff
@@ -270,16 +360,26 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
...
+      }
+      
         return cell
     }
     
     override open func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        
         let cell = tableView.cellForRow(at: indexPath) as! EPContactCell
         let selectedContact =  cell.contact!
+      if selectedContact.firstName == "+ Add phone number" {
```
Comment (danageva) [inline Pods/EPContactsPicker.swift]: indentation

File: Pods/EPContactsPicker.swift [hunk 10]
```diff
@@ -288,10 +388,14 @@ open class EPContactsPicker: UITableViewController, UISearchResultsUpdating, UIS
                     return selectedContact.contactId != $0.contactId
                 }
             }
-            else {
+            else if (self.multiSelectContactLimit == 0 || self.selectedContacts.count < Int(self.multiSelectContactLimit)) {
                 cell.accessoryType = UITableViewCellAccessoryType.checkmark
+                cell.tintColor = EPGlobalConstants.Colors.nxYellow
                 selectedContacts.append(selectedContact)
             }
+          let enabled = self.multiSelectContactLimit == 0
```
Comment (danageva) [inline Pods/EPContactsPicker.swift:394]: I think that you don't need to compare multiSelectContactLimit to 0

File: Pods/EPExtensions.swift [hunk 11]
```diff
@@ -31,3 +31,28 @@ extension String {
         })
     }
 }
+
+extension UIAlertController {
+  func changeFont(view:UIView, font:UIFont) {
```
Comment (danageva) [inline Pods/EPExtensions.swift]: updateFont(for view: UIView, to font: UIFont)...

**Review by danageva — CHANGES_REQUESTED** [review]

File: Pods/EPContactsPicker.swift (hunk 3 above)
Comment (hihilla) [reply Pods/EPContactsPicker.swift]: this is not the "add contact" method, this is presenting an alert when user did not give permissions for contacts

Comment (hihilla) [reply Pods/EPContactsPicker.swift]: maybe I can change it to presentPermissionsAlert?

File: Pods/EPContactsPicker.swift (hunk 10 above)
Comment (hihilla) [reply Pods/EPContactsPicker.swift:394]: self.multiSelectContactLimit == 0 means no limit
I didn't want to change the general contact picker logic

**Pull request merged**